openage/convert/interface/interfacecutter.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2016-2016 the openage authors. See copying.md for legal info.
+
+""" Additional processing of the user interface assets """
+
+import os
+import re
+import tempfile
+from subprocess import Popen, PIPE
+
+from ..texture import TextureImage
+
+INTERFACE_INGAME_HUD_BACKGROUNDS = [
+    51141,
+    51142,
+    51143,
+    51144,
+    51145,
+    51146,
+    51147,
+    51148,
+    51149,
+    51150,
+    51151,
+    51152,
+    51153,
+    51154,
+    51157,
+    51158,
+    51159,
+    51160,
+]
+
+TOP_STRIP_PATTERN_XYWH = (400, 0, 64, 32)
+
+KNOWN_SUBTEXTURE_CORNER_COORDS = [
+    (0, 806, 339, 1024),
+    (850, 806, 1280, 1024),
+    (8, 10, 34, 27),
+    (85, 10, 111, 27),
+    (162, 10, 188, 27),
+    (239, 10, 265, 27),
+    (316, 10, 342, 27),
+]
+
+
+def is_ingame_hud_background(filename):
+    """
+    True if in the hardcoded list of the known ingame hud backgrounds.
+    """
+    try:
+        basename = re.sub(r"^interface/(\d+)\.slp$", r'\1', filename)
+        INTERFACE_INGAME_HUD_BACKGROUNDS.index(basename)
+        return True
+    except ValueError:
+        return False
+
+
+class InterfaceCutter(object):
+    """
+    Callable object that cuts interface textures.
+    """
+
+    def __init__(self, filename):
+        self.filename = filename
+
+    def __call__(self, image):
+        if is_ingame_hud_background(self.filename):
+            pil_image = image.get_pil_image()
+
+            yield self.__cut_top_strip(pil_image)
+
+            for coords in KNOWN_SUBTEXTURE_CORNER_COORDS:
+                yield TextureImage(pil_image.crop(coords))
+        else:
+            yield image
+
+    def __cut_top_strip(self, pil_image):
+        """
+        Finds a horisontally tilable piece of the strip that's at the top of the HUD.
+        """
+
+        try:
+            uncut = tempfile.NamedTemporaryFile(delete=False)
+            uncut.close()
+
+            pil_image.save(uncut.name, "PNG")
+
+            seamless_top_strip_tile_search = Popen([
+                'python',
+                os.path.join(os.path.dirname(os.path.realpath(__file__)), 'visgrep.py'),
+                '-J', str(TOP_STRIP_PATTERN_XYWH[1] + TOP_STRIP_PATTERN_XYWH[3]),
+                '-u', str(TOP_STRIP_PATTERN_XYWH[0]),
+                '-v', str(TOP_STRIP_PATTERN_XYWH[1]),
+                '-M', str(TOP_STRIP_PATTERN_XYWH[2]),
+                '-N', str(TOP_STRIP_PATTERN_XYWH[3]),
+                '-W', str(TOP_STRIP_PATTERN_XYWH[2]),
+                '-H', str(TOP_STRIP_PATTERN_XYWH[3]),
+                '-t', '100000',
+                uncut.name, uncut.name, uncut.name], stdout=PIPE)
```

Comment:
No, there's no need for a py interface. I mean it's still possible implementing in a way it can either be used via commandline or by importing and using it, which would even improve the "dependency we had to rewrite" by extending it with a nicer API interface.

Pulling OpenCV instead of that might be a bit heavy.